src/libs/actions/Report.js

```diff
@@ -226,6 +265,14 @@ function getSimplifiedReportObject(report) {
         visibility,
         isOwnPolicyExpenseChat: lodashGet(report, ['isOwnPolicyExpenseChat'], false),
     };
+
+    const currentMaxSeqNumber = lodashGet(allReports, [report.reportID, 'maxSequenceNumber'], 0);
```

Comment:
Here we are getting the maxSequenceNumber from the storage. Which means we expect it to be present. Ignore default value for instance. 

But then we early return on line https://github.com/Expensify/App/blob/f69dd30c47946e561888b7ed781c4f9d59cdf008/src/libs/actions/Report.js#L169-L172 

these statements contradict in nature. why? the program should not reach that point where value does not exist.

Apart from it, we should not pollute this function at all.  This is a side-effect. Instead of doing these changes move them to https://github.com/Expensify/App/blob/f69dd30c47946e561888b7ed781c4f9d59cdf008/src/libs/actions/Report.js#L429

--- 
Please add more inline comments to explain the tricky pieces of logic.